libmozdata/vcs_map.py

```diff
@@ -0,0 +1,112 @@
+import argparse
+import os
+import sys
+import urllib.request
+
+import hglib
+import requests
+from tqdm import tqdm
+
+MAPPER_SERVICE = "https://mapper.mozilla-releng.net"
+VCS_MAP_FULL_PATH = "vcs_map_full"
+VCS_MAP_CACHE_PATH = "vcs_map_cache"
+
+git_to_mercurial_mapping = {}
+mercurial_to_git_mapping = {}
+
+
+def download_mapfile():
+    if not os.path.exists(VCS_MAP_FULL_PATH):
+        urllib.request.urlretrieve(
+            f"{MAPPER_SERVICE}/gecko-dev/mapfile/full", VCS_MAP_FULL_PATH
+        )
+
+
+MAPFILE_LOADED = False
+
+
+def load_mapfile():
+    global MAPFILE_LOADED
+    if MAPFILE_LOADED:
+        return
+
+    MAPFILE_LOADED = True
+
+    def _read_mapping_file(path):
+        with open(path, "r") as f:
+            for line in f:
```

Comment:
why not using `f.readlines()` and avoiding the split below ?

Reply:
The split is necessary as the file has this format:
```
GIT_HASH1 MERCURIAL_HASH1
GIT_HASH2 MERCURIAL_HASH2
GIT_HASH3 MERCURIAL_HASH3
```

Reply:
So each line must be splitted by " " to have the git hash and the mercurial hash.

Reply:
Best i have is :

```python
lines = open(path).read().splitlines()
for git_hash, mercurial_hash in map(lambda x: x.split(' '), lines):
   ...
```